monkey/agent_plugins/exploiters/rdp/src/rdp_credentials_generator.py

```diff
@@ -9,35 +9,55 @@
 
 
 def generate_rdp_credentials(
-    credentials: Sequence[Credentials], domains: Sequence[str]
+    credentials: Sequence[Credentials], domains: Sequence[str], running_from_windows: bool
 ) -> Sequence[Credentials]:
     brute_force_credentials = generate_brute_force_credentials(
         credentials,
         identity_filter=identity_type_filter([Username]),
         secret_filter=secret_type_filter([Password, LMHash, NTHash]),
     )
-    rdp_credentials = list(_add_domains_to_usernames(brute_force_credentials, domains))
+    rdp_credentials = list(
+        _add_domains_to_usernames(brute_force_credentials, domains, running_from_windows)
+    )
 
     return _remove_duplicate_credentials(rdp_credentials)
 
 
 def _add_domains_to_usernames(
-    credentials: Sequence[Credentials], domains: Sequence[str]
+    credentials: Sequence[Credentials], domains: Sequence[str], running_from_windows: bool
 ) -> Iterable[Credentials]:
+    local_domains = _get_domains(running_from_windows)
+    all_domains = [*domains, *local_domains]
+
     for credential in credentials:
         if credential.identity is None:
             continue
 
         if "\\" in credential.identity.username:
             yield credential
 
-        for domain in domains:
+        for domain in all_domains:
             yield Credentials(
                 identity=Username(username=f"{domain}\\{credential.identity.username}"),
                 secret=credential.secret,
             )
 
 
+def _get_domains(running_from_windows: bool) -> Set[str]:
+    domains = {"."}
+
+    if running_from_windows:
+        domains.add(_get_domain())
+
+    return domains
+
+
+def _get_domain() -> str:
```

Comment:
These functions could use better names.
```suggestion
def _get_local_machine_domain() -> str:
```